src/course-unit/CourseUnit.test.jsx

```diff
@@ -668,4 +686,389 @@ describe('<CourseUnit />', () => {
+              block_type: 'drag-and-drop-v2',
+            },
+          ],
+        });
+
+      await executeThunk(fetchCourseVerticalChildrenData(blockId), store.dispatch);
+      expect(getAllByTestId('course-xblock')).toHaveLength(3);
+    });
+
+    it('should display the "Paste component" button after copying a xblock to clipboard', async () => {
+      const { getByRole, getAllByLabelText } = render(<RootWrapper />);
+
+      await waitFor(() => {
+        const [xblockActionBtn] = getAllByLabelText(courseXBlockMessages.blockActionsDropdownAlt.defaultMessage);
+        userEvent.click(xblockActionBtn);
+        userEvent.click(getByRole('button', { name: courseXBlockMessages.blockLabelButtonCopyToClipboard.defaultMessage }));
+      });
+
+      axiosMock
+        .onGet(getCourseSectionVerticalApiUrl(blockId))
+        .reply(200, {
+          ...courseSectionVerticalMock,
+          user_clipboard: clipboardXBlock,
+        });
+
+      await executeThunk(fetchCourseSectionVerticalData(blockId), store.dispatch);
+      await executeThunk(copyToClipboard(blockId), store.dispatch);
+      expect(getByRole('button', { name: pasteComponentMessages.pasteComponentButtonText.defaultMessage })).toBeInTheDocument();
+    });
+
+    it('should copy a unit, paste it as a new unit, and update the course section vertical data', async () => {
+      const {
+        getAllByTestId, getByTestId, getByRole, getByText,
+      } = render(<RootWrapper />);
+
+      axiosMock
+        .onGet(getCourseUnitApiUrl(courseId))
+        .reply(200, {
+          ...courseUnitIndexMock,
+          enable_copy_paste_units: true,
+        });
+
+      await executeThunk(fetchCourseUnitQuery(courseId), store.dispatch);
+
+      userEvent.click(getByRole('button', { name: sidebarMessages.actionButtonCopyUnitTitle.defaultMessage }));
+
+      axiosMock
+        .onGet(getCourseSectionVerticalApiUrl(blockId))
+        .reply(200, {
+          ...courseSectionVerticalMock,
+          user_clipboard: clipboardUnit,
+        });
+
+      await executeThunk(fetchCourseSectionVerticalData(blockId), store.dispatch);
+      await executeThunk(copyToClipboard(blockId), store.dispatch);
+
+      userEvent.click(getByTestId('dropdown-paste-unit'));
+
+      userEvent.click(getByText(courseSequenceMessages.pasteAsNewUnitLink.defaultMessage));
+
+      let units = null;
+      const updatedCourseSectionVerticalData = cloneDeep(courseSectionVerticalMock);
+      const updatedAncestorsChild = updatedCourseSectionVerticalData.xblock_info.ancestor_info.ancestors[0];
+      set(updatedCourseSectionVerticalData, 'xblock_info.ancestor_info.ancestors[0].child_info.children', [
+        ...updatedAncestorsChild.child_info.children,
+        courseUnitMock,
+      ]);
+
+      units = getAllByTestId('course-unit-btn');
+      const courseUnits = courseSectionVerticalMock.xblock_info.ancestor_info.ancestors[0].child_info.children;
+      expect(units.length).toEqual(courseUnits.length);
```

Comment:
[optional] this case and below can be replaced with
```suggestion
      expect(units).toHaveLength(courseUnits.length);
```